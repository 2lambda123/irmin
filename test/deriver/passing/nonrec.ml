(* Ensure that 'nonrec' assertions are respected *)
module S1 : sig
  type nonrec t = t list [@@deriving irmin]

  type nonrec t_alias = t_alias list [@@deriving irmin]
end = struct
  type nonrec t = t list [@@deriving irmin]

  type nonrec t_alias = t_alias list [@@deriving irmin]
end

(* Now test the interaction of 'nonrec' with custom naming *)
module S2 : sig
  type nonrec t = t list [@@deriving irmin { name = "t_witness" }]

  type nonrec t_alias = t_alias list [@@deriving irmin { name = "t_witness" }]
end = struct
  type nonrec t = t list [@@deriving irmin { name = "t_witness" }]

  type nonrec t_alias = t_alias list [@@deriving irmin { name = "t_witness" }]
end
