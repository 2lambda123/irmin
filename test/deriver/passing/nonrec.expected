module S1 :
  sig
    type nonrec t = t list[@@deriving irmin]
    val t : t Irmin.Type.t
    type nonrec t_alias = t_alias list[@@deriving irmin]
    val t_alias_t : t_alias Irmin.Type.t
  end =
  struct
    type nonrec t = t list[@@deriving irmin]
    let t = let open Irmin.Type in list t
    type nonrec t_alias = t_alias list[@@deriving irmin]
    let t_alias_t = let open Irmin.Type in list t_alias_t
  end 
module S2 :
  sig
    type nonrec t = t list[@@deriving irmin { name = "t_witness" }]
    val t_witness : t Irmin.Type.t
    type nonrec t_alias = t_alias list[@@deriving
                                        irmin { name = "t_witness" }]
    val t_witness : t_alias Irmin.Type.t
  end =
  struct
    type nonrec t = t list[@@deriving irmin { name = "t_witness" }]
    let t_witness = let open Irmin.Type in list t
    type nonrec t_alias = t_alias list[@@deriving
                                        irmin { name = "t_witness" }]
    let t_witness = let open Irmin.Type in list t_alias_t
  end 
